test-data/unit/check-inference-context.test

```diff
@@ -569,6 +569,15 @@ class A: pass
 class B: pass
 [builtins fixtures/list.pyi]
 
+[case testInferLambdaTypeUsingContext]
+x = (lambda x: x + 1)(1)  # type: str
+reveal_type((lambda x, y: x + y)(1, 2))  # E: Revealed type is 'builtins.int'
+(lambda x, y: x + y)(1, "")  # E: Unsupported operand types for + ("int" and "str")
+(lambda *, x, y: x + y)(x=1, y="")  # E: Unsupported operand types for + ("int" and "str")
+[out]
+main:1: error: Incompatible types in assignment (expression has type "int", variable has type "str")
+main:1: error: Incompatible return value type (got "int", expected "str")
```

Comment:
Because we check the lambda against the full context, including the return type, and it fails to check. Then we check the call expression against the variable, which again fails. Maybe it is possible to check the lambda with return type defined to be `object`